### DISCUSSION TITLE
How could I go about building a simple pulse counting circuit?

### DISCUSSION THREAD

**Discussion by jtomes123** [ozel/DIY_particle_detector#55] [Q&A]
Hi, I finally managed to build the circuit. I was thinking that I would like a basic pulse counting circuit just for checking if object a is more active than object B.

What I came up with is using an opamp as a comparator, to get a high every time there is a pulse, then I can use a transistor or an optocoupler to connect it to either a microcontroller or some additional circuitry using e. g. the cd4017 to count the pulses. However my electronics knowledge is still limited, so there might be an easier way to do this.

**Comment by ozel** [comment]
Hi Jakub, Your idea of using a comparator followed by a counting chip is exactly the right approach with discrete electronics.

Note that you can, of course, also count the pulses with a microcontroller's ADC. In that case, whether the pulses stick out of the noise depends on the ADC input impedance, resolution, and selectable voltage ranges of the ADC. The sampling rate should be about 50 kHz if possible. I can be lower just for counting, but the debugging of what's going on might be harder.
Some microcontrollers also feature built-in analogue comparator inputs that may be good enough for counting.

I have long intended to publish a reference microcontroller setup, but I have not had sufficient time yet.
It's definitely doable,  related discussion on this here: https://github.com/ozel/DIY_particle_detector/discussions/35
